dist/tools/ci/can_fast_ci_run.py

```diff
@@ -164,14 +164,39 @@ def classify_changes(riotbase=None, upstream_branch="master"):
             match = REGEX_GIT_DIFF_SINGLE_FILE.match(line)
             if match:
                 file = match.group(1)
-                change_set.add_file(file)
+
+                if only_comment_change(file, upstream_branch) is False:
+                    change_set.add_file(file)
+
                 continue
 
             raise Exception("Failed to parse \"{}\"".format(line))
 
     return change_set
 
 
+def get_hash_without_comments(file, ref):
+    result = subprocess.run(
+        f"git show {ref}:{file} | gcc -fpreprocessed -dD -E -P - | md5sum",
+        shell=True,
+        capture_output=True,
+        check=True,
```

Comment:
Since you set `check=True` and according to [the documentation of subprocess.run](https://docs.python.org/3/library/subprocess.html#subprocess.run), if the process returns with a non zero code, the `CalledProcessError` is raised.

Only subprocess.run is expected to raise an exception, so the try/except should be placed around this call, not around the whole content of `only_comment_change`:

```python
def get_hash_without_comments(file, ref):
    try:
        result = subprocess.run(
            f"git show {ref}:{file} | gcc -fpreprocessed -dD -E -P - | md5sum",
            shell=True,
            capture_output=True,
            check=True,
        )
    except CalledProcessError:
        print_err(f"Cannot get hash without comments for file '{file}' on ref '{ref}')
        return
    return result.stdout
```

And `only_comment_change` would become:

```python
def only_comment_change(file, upstream_branch):
    if file[-2:] not in {".c", ".h"}:
        return False
    hash_head = get_hash_without_comments(file, "HEAD")
    if hash_head is None:
        return False
    hash_upstream_branch = get_hash_without_comments(file, upstream_branch)
    if hash_upstream_branch is None:
        return False

    return hash_head == hash_upstream_branch
```